appserver/admingui/plugin-service/src/main/java/org/glassfish/admingui/plugin/ConsolePluginService.java

```diff
@@ -80,24 +81,39 @@ protected synchronized void init() {
         if ((providers != null) && (providers.iterator().hasNext())) {
             // Get our parser...
             ConfigParser parser = new ConfigParser(habitat);
-            URL url = null;
             String id = null;
 
+            Map<URL, ClassLoader> configUrls = new HashMap<>();
+
             // Loop through the configs and add them all
             for (ConsoleProvider provider : providers) {
                 // Read the contents from the URL
-                url = provider.getConfiguration();
-                if (url == null) {
-                    url = provider.getClass().getClassLoader().getResource(
-                            ConsoleProvider.DEFAULT_CONFIG_FILENAME);
+                URL url = provider.getConfiguration();
+                Iterator<URL> urlsIterator;
+                if (url != null) {
+                    urlsIterator = List.of(url).iterator();
+                } else {
+                    try {
+                        urlsIterator = provider.getClass().getClassLoader().getResources(
+                                ConsoleProvider.DEFAULT_CONFIG_FILENAME).asIterator();
+                    } catch (IOException ex) {
+                        throw new RuntimeException(ex);
```

Comment:
You throw `RuntimeException` while original code not. Why?

Reply:
This exception should not happen. If it does, it's a bug and it's thrown as a RuntimeException, so that it cancels all processing and is logged in the server log. I didn't want to just log it and ignore, because if it really happens it's a serious bug and not something that can be ignored.

Is it an antipattern? Do you have a better idea how to address this?

Reply:
No, this is not an antipattern. You're right, if an exception throws here, then something extraordinary has happened.